photologue/utils/zipfile.py

```diff
@@ -0,0 +1,114 @@
+    if gallery:
+        logger.debug('Using pre-existing gallery.')
+    else:
+        logger.debug(
+            force_text('Creating new gallery "{0}".').format(title))
+        gallery = Gallery.objects.create(title=title,
+                                         slug=slugify(title),
+                                         description=description,
+                                         is_public=is_public)
+        gallery.sites.add(current_site)
+
+    for filename in sorted(zip.namelist()):
+
+        logger.debug('Reading file "{}".'.format(filename))
+
+        if filename.startswith('__') or filename.startswith('.'):
+            logger.debug('Ignoring file "{}".'.format(filename))
+            continue
+
+        if os.path.dirname(filename):
+            logger.warning('Ignoring file "{}" as it is in a subfolder; all images should be in the top '
+                           'folder of the zip.'.format(filename))
+            if request:
+                messages.warning(request,
+                                 _('Ignoring file "{filename}" as it is in a subfolder; all images should '
+                                   'be in the top folder of the zip.').format(filename=filename),
+                                 fail_silently=True)
+            continue
+
+        data = zip.read(filename)
+
+        if not len(data):
+            logger.debug('File "{}" is empty.'.format(filename))
+            continue
+
+        photo_title_root = title if title else gallery.title
+
+        # A photo might already exist with the same slug. So it's somewhat inefficient,
+        # but we loop until we find a slug that's available.
+        while True:
+            photo_title = ' '.join([photo_title_root, str(count)])
+            slug = slugify(photo_title)
+            if Photo.objects.filter(slug=slug).exists():
+                count += 1
+                continue
+            break
+
+        photo = Photo(title=photo_title,
+                      slug=slug,
+                      caption=caption,
+                      is_public=is_public)
+
+        # Basic check that we have a valid image.
+        try:
+            file = BytesIO(data)
+            opened = Image.open(file)
+            opened.verify()
+        except Exception:
+            # Pillow doesn't recognize it as an image.
+            # If a "bad" file is found we just skip it.
+            # But we do flag this both in the logs and to the user.
+            logger.error('Could not process file "{}" in the .zip archive.'.format(
+                filename))
+            if request:
+                messages.warning(request,
+                                 _('Could not process file "{0}" in the .zip archive.').format(
+                                     filename),
+                                 fail_silently=True)
+            continue
+
+        contentfile = ContentFile(data)
+        photo.image.save(filename, contentfile)
+
+        if USE_CELERY:
```

Comment:
Rather than have a name that's specific (CELERY) I'd rather have a name that's more generic (if next year we decide to use for example RQ). For example PROCESSING_MODE with a default value of "synchronous" and the other possible value defined in settings would be "celery".